experimental/adaptive-tool/server/lg/src/providers/foldingRange.ts

```diff
@@ -0,0 +1,55 @@
+import { FoldingRangeParams, FoldingRange, TextDocuments } from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import * as path from 'path';
+import * as util from '../util';
+
+export function foldingRange(params: FoldingRangeParams, documents: TextDocuments<TextDocument>) {
+	const document = documents.get(params.textDocument.uri)!;
+	const item: FoldingRange[] = [];
+	if (!util.isLgFile(path.basename(document.uri))) {
+		return item;
+	}
+	const lineCount = document.lineCount;
+	for (var i = 0;i < lineCount;i++) {
```

Comment:
it will be better to use `let i = 0` in the for loop. Due the function scope.

Reply:
fixed